File: app/customssiv/src/main/java/me/devsaki/hentoid/customssiv/ImageViewState.java
package me.devsaki.hentoid.customssiv;

import android.graphics.PointF;

import androidx.annotation.NonNull;

import java.io.Serializable;

/**
 * Wraps the scale, center and orientation of a displayed image for easy restoration on screen rotate.
 */
@SuppressWarnings("WeakerAccess")
public class ImageViewState implements Serializable {

    private final float scale;
    private final float virtualScale;
    private final float centerX;
    private final float centerY;
    private final int orientation;

    public ImageViewState(float scale, float virtualScale, @NonNull PointF center, int orientation) {
        this.scale = scale;
        this.virtualScale = virtualScale;
        this.centerX = center.x;
        this.centerY = center.y;
        this.orientation = orientation;
    }

    public float getScale() {
        return scale;
    }

    public float getVirtualScale() {
        return virtualScale;
    }

    @NonNull
    public PointF getCenter() {
        return new PointF(centerX, centerY);
    }

    public int getOrientation() {
        return orientation;
    }

}

File: app/customssiv/src/main/java/me/devsaki/hentoid/customssiv/ImageViewState.kt
package me.devsaki.hentoid.customssiv

import android.graphics.PointF
import me.devsaki.hentoid.customssiv.CustomSubsamplingScaleImageView.Orientation
import java.io.Serializable

/**
 * Wraps the scale, center and orientation of a displayed image for easy restoration on screen rotate.
 */
class ImageViewState(
    private var scale: Float,
    private var virtualScale: Float,
    center: PointF,
    private var orientation: Orientation
) : Serializable {
    private var centerX = 0f
    private var centerY = 0f

    init {
        this.centerX = center.x
        this.centerY = center.y
    }

    fun getScale(): Float {
        return scale
    }

    fun getVirtualScale(): Float {
        return virtualScale
    }

    fun getCenter(): PointF {
        return PointF(centerX, centerY)
    }

    fun getOrientation(): CustomSubsamplingScaleImageView.Orientation {
        return orientation
    }
}